app/components/TuneCard/index.js

```diff
@@ -74,9 +80,15 @@ export function TuneCard({ maxwidth, artistName, collectionName, cardImg, previe
           </If>
         }
       />
-      <audio ref={audioRef} data-testid="audio" src={previewUrl}></audio>
+      <audio ref={audioRef} data-testid="audio"></audio>
     </ItemCard>
   );
+
+  return (
+    <If condition={songId} otherwise={renderItemCard()}>
```

Comment:
The renderItemCard function in both if and otherwise block is same so that we don't repeat the card item code. If songId is present, we wrap the CardItem with a Link comp else render only the cardItem.

Reply:
Honestly, that feels like a limitation. The detail page for all purposes considered must be rendered by the `ITunesDetail` container. The view of the detail page is fundamentally different from the trackcard. The coupling you have is not the best way.